crypto/evp/m_sigver.c

```diff
@@ -59,8 +59,12 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
         ctx->provctx = NULL;
     }
 
-    if (ctx->pctx == NULL)
-        ctx->pctx = EVP_PKEY_CTX_new(pkey, e);
+    if (ctx->pctx == NULL) {
+        if (libctx != NULL)
+            ctx->pctx = EVP_PKEY_CTX_new_from_pkey(libctx, pkey, props);
+        else
+            ctx->pctx = EVP_PKEY_CTX_new(pkey, e);
+    }
```

Comment:
And what do you do in the `ctx->pctx != NULL && ctx->pctx->libctx != NULL && libctx != NULL` case?  Why is one chosen over the other?  Could that be problematic?

Reply:
The libctx from the existing pctx takes precedence. From the docs:

````
If the EVP_PKEY_CTX to be used is
created by EVP_DigestSignInit_ex then it will use the B<OPENSSL_CTX> specified
in I<libctx> and the property query string specified in I<props>.
````
So...libctx is only used if a pctx is created.

Reply:
Hmmm, ok.

Reply:
I am not sure in this case whether putting the libctx as first parameter isn't misleading. Basically as it is only auxiliary parameter which is not always used, it should not be the first parameter.